### ISSUE TITLE
lib/private/Share20/DefaultShareProvider.php: More than 1000 expressions in a list are not allowed on Oracle.

### ISSUE THREAD

**Issue by nickvergessen** [nextcloud/server#32593]
```json
{
  "reqId": "aX0KMet70L3wHH2VMJku",
  "level": 3,
  "time": "2022-05-25T11:47:59+00:00",
  "remoteAddr": "…",
  "user": "…",
  "app": "core",
  "method": "PROPFIND",
  "url": "/remote.php/dav/files/…/Talk",
  "message": "More than 1000 expressions in a list are not allowed on Oracle.",
  "userAgent": "…",
  "version": "24.0.1.1",
  "exception": {
    "Exception": "Doctrine\\DBAL\\Query\\QueryException",
    "Message": "More than 1000 expressions in a list are not allowed on Oracle.",
    "Code": 0,
    "Trace": [
      {
        "file": "/var/www/html/lib/private/Share20/DefaultShareProvider.php",
        "line": 691,
        "function": "execute",
        "class": "OC\\DB\\QueryBuilder\\QueryBuilder",
        "type": "->"
      },
      {
        "file": "/var/www/html/lib/private/Share20/Manager.php",
        "line": 1310,
        "function": "getSharesInFolder",
        "class": "OC\\Share20\\DefaultShareProvider",
        "type": "->"
      },
      {
        "function": "OC\\Share20\\{closure}",
        "class": "OC\\Share20\\Manager",
        "type": "->",
        "args": [
          "*** sensitive parameters replaced ***"
        ]
      },
      {
        "file": "/var/www/html/lib/private/Share20/Manager.php",
        "line": 1319,
        "function": "array_reduce"
      },
      {
        "file": "/var/www/html/apps/dav/lib/Connector/Sabre/SharesPlugin.php",
        "line": 149,
        "function": "getSharesInFolder",
        "class": "OC\\Share20\\Manager",
        "type": "->"
      },
      {
        "file": "/var/www/html/apps/dav/lib/Connector/Sabre/SharesPlugin.php",
        "line": 206,
        "function": "getSharesFolder",
        "class": "OCA\\DAV\\Connector\\Sabre\\SharesPlugin",
        "type": "->"
      },
      {
        "file": "/var/www/html/3rdparty/sabre/event/lib/WildcardEmitterTrait.php",
        "line": 89,
        "function": "handleGetProperties",
        "class": "OCA\\DAV\\Connector\\Sabre\\SharesPlugin",
        "type": "->"
      },
      {
        "file": "/var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line": 1052,
        "function": "emit",
        "class": "Sabre\\DAV\\Server",
        "type": "->"
      },
      {
        "file": "/var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line": 984,
        "function": "getPropertiesByNode",
        "class": "Sabre\\DAV\\Server",
        "type": "->"
      },
      {
        "file": "/var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line": 1662,
        "function": "getPropertiesIteratorForPath",
        "class": "Sabre\\DAV\\Server",
        "type": "->"
      },
      {
        "file": "/var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line": 1647,
        "function": "writeMultiStatus",
        "class": "Sabre\\DAV\\Server",
        "type": "->"
      },
      {
        "file": "/var/www/html/3rdparty/sabre/dav/lib/DAV/CorePlugin.php",
        "line": 346,
        "function": "generateMultiStatus",
        "class": "Sabre\\DAV\\Server",
        "type": "->"
      },
      {
        "file": "/var/www/html/3rdparty/sabre/event/lib/WildcardEmitterTrait.php",
        "line": 89,
        "function": "httpPropFind",
        "class": "Sabre\\DAV\\CorePlugin",
        "type": "->"
      },
      {
        "file": "/var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line": 472,
        "function": "emit",
        "class": "Sabre\\DAV\\Server",
        "type": "->"
      },
      {
        "file": "/var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line": 253,
        "function": "invokeMethod",
        "class": "Sabre\\DAV\\Server",
        "type": "->"
      },
      {
        "file": "/var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line": 321,
        "function": "start",
        "class": "Sabre\\DAV\\Server",
        "type": "->"
      },
      {
        "file": "/var/www/html/apps/dav/lib/Server.php",
        "line": 352,
        "function": "exec",
        "class": "Sabre\\DAV\\Server",
        "type": "->"
      },
      {
        "file": "/var/www/html/apps/dav/appinfo/v2/remote.php",
        "line": 35,
        "function": "exec",
        "class": "OCA\\DAV\\Server",
        "type": "->"
      },
      {
        "file": "/var/www/html/remote.php",
        "line": 166,
        "args": [
          "/var/www/html/apps/dav/appinfo/v2/remote.php"
        ],
        "function": "require_once"
      }
    ],
    "File": "/var/www/html/lib/private/DB/QueryBuilder/QueryBuilder.php",
    "Line": 264,
    "CustomMessage": "More than 1000 expressions in a list are not allowed on Oracle."
  }
```

**Comment by schiessle** [comment]
I saw today the same error on a Nextcloud 28.0.7.4 on cadav requests:

```
Assistent
Bildgenerirung{
  "reqId": "h4GkaBMxZBbaYGvGCNf6",
  "level": 3,
  "time": "2024-07-12T17:16:45+00:00",
  "remoteAddr": "2a0f:6480:1::2",
  "user": "abb8c2f2-26c2-1034-821f-c3ab67df07dc",
  "app": "core",
  "method": "REPORT",
  "url": "/remote.php/caldav/calendars/abb8c2f2-26c2-1034-821f-c3ab67df07dc/app-generated--deck--board-27/",
  "message": "More than 1000 expressions in a list are not allowed on Oracle.",
  "userAgent": "Evolution/3.46.4",
  "version": "28.0.7.4",
  "exception": {
    "Exception": "Doctrine\\DBAL\\Query\\QueryException",
    "Message": "More than 1000 expressions in a list are not allowed on Oracle.",
    "Code": 0,
    "Trace": [
      {
        "file": "/var/www/nextcloud/lib/private/DB/QueryBuilder/QueryBuilder.php",
        "line": 293,
        "function": "execute",
        "class": "OC\\DB\\QueryBuilder\\QueryBuilder",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/lib/public/AppFramework/Db/QBMapper.php",
        "line": 335,
        "function": "executeQuery",
        "class": "OC\\DB\\QueryBuilder\\QueryBuilder",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/apps/deck/lib/Db/LabelMapper.php",
        "line": 92,
        "function": "findEntities",
        "class": "OCP\\AppFramework\\Db\\QBMapper",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/apps/deck/lib/Service/CardService.php",
        "line": 146,
        "function": "findAssignedLabelsForCards",
        "class": "OCA\\Deck\\Db\\LabelMapper",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/apps/deck/lib/Service/CardService.php",
        "line": 207,
        "function": "enrichCards",
        "class": "OCA\\Deck\\Service\\CardService",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/apps/deck/lib/DAV/DeckCalendarBackend.php",
        "line": 85,
        "function": "findCalendarEntries",
        "class": "OCA\\Deck\\Service\\CardService",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/apps/deck/lib/DAV/Calendar.php",
        "line": 152,
        "function": "getChildren",
        "class": "OCA\\Deck\\DAV\\DeckCalendarBackend",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/apps/deck/lib/DAV/Calendar.php",
        "line": 135,
        "function": "getBackendChildren",
        "class": "OCA\\Deck\\DAV\\Calendar",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/apps/deck/lib/DAV/Calendar.php",
        "line": 90,
        "function": "getChildren",
        "class": "OCA\\Deck\\DAV\\Calendar",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/3rdparty/sabre/dav/lib/CalDAV/Plugin.php",
        "line": 598,
        "function": "calendarQuery",
        "class": "OCA\\Deck\\DAV\\Calendar",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/3rdparty/sabre/dav/lib/CalDAV/Plugin.php",
        "line": 258,
        "function": "calendarQueryReport",
        "class": "Sabre\\CalDAV\\Plugin",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/3rdparty/sabre/event/lib/WildcardEmitterTrait.php",
        "line": 89,
        "function": "report",
        "class": "Sabre\\CalDAV\\Plugin",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/CorePlugin.php",
        "line": 690,
        "function": "emit",
        "class": "Sabre\\DAV\\Server",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/3rdparty/sabre/event/lib/WildcardEmitterTrait.php",
        "line": 89,
        "function": "httpReport",
        "class": "Sabre\\DAV\\CorePlugin",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line": 472,
        "function": "emit",
        "class": "Sabre\\DAV\\Server",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line": 253,
        "function": "invokeMethod",
        "class": "Sabre\\DAV\\Server",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line": 321,
        "function": "start",
        "class": "Sabre\\DAV\\Server",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/apps/dav/appinfo/v1/caldav.php",
        "line": 123,
        "function": "exec",
        "class": "Sabre\\DAV\\Server",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/remote.php",
        "line": 172,
        "args": [
          "/var/www/nextcloud/apps/dav/appinfo/v1/caldav.php"
        ],
        "function": "require_once"
      }
    ],
    "File": "/var/www/nextcloud/lib/private/DB/QueryBuilder/QueryBuilder.php",
    "Line": 263,
    "message": "More than 1000 expressions in a list are not allowed on Oracle.",
    "query": "SELECT `l`.*, `card_id` FROM `*PREFIX*deck_labels` `l` INNER JOIN `*PREFIX*deck_assigned_labels` `al` ON l.id = al.label_id WHERE `card_id` IN (:dcValue1) ORDER BY `l`.`id` ASC",
    "exception": {},
    "CustomMessage": "More than 1000 expressions in a list are not allowed on Oracle."
  }
}

```

**Comment by joshtrichards** [comment]
Fixed in #32185
